src/content-tags-drawer/ContentTagsDrawer.test.jsx

```diff
@@ -30,18 +37,29 @@ jest.mock('./data/apiHooks', () => ({
   })),
 }));
 
-jest.mock('../taxonomy/data/apiHooks', () => ({
-  useTaxonomyListDataResponse: jest.fn(),
-  useIsTaxonomyListDataLoaded: jest.fn(),
+jest.mock('../taxonomy/data/api', () => ({
+  // By default, the mock taxonomy list will never load (promise never resolves):
+  getTaxonomyListData: jest.fn(),
 }));
 
+const queryClient = new QueryClient();
+
 const RootWrapper = () => (
   <IntlProvider locale="en" messages={{}}>
-    <ContentTagsDrawer />
+    <QueryClientProvider client={queryClient}>
+      <ContentTagsDrawer />
+    </QueryClientProvider>
   </IntlProvider>
 );
 
 describe('<ContentTagsDrawer />', () => {
+  beforeEach(async () => {
+    await queryClient.resetQueries();
+    // By default, we mock the API call with a promise that never resolves.
+    // You can override this in specific test.
+    getTaxonomyListData.mockReturnValue(new Promise(() => {}));
+  });
+
   it('should render page and page title correctly', () => {
     const { getByText } = render(<RootWrapper />);
```

Comment:
@bradenmacdonald Maybe we should also create a task to refactor the code following [good practices](https://forum.opencraft.com/t/some-tips-and-tricks-for-testing-react-code/1657)